components/suggest/src/schema.rs

```diff
@@ -149,15 +153,35 @@ impl ConnectionInitializer for SuggestConnectionInitializer {
         Ok(db.execute_batch(SQL)?)
     }
 
-    fn upgrade_from(&self, _db: &Transaction<'_>, version: u32) -> open_database::Result<()> {
+    fn upgrade_from(&self, tx: &Transaction<'_>, version: u32) -> open_database::Result<()> {
         match version {
             1..=15 => {
                 // Treat databases with these older schema versions as corrupt,
                 // so that they'll be replaced by a fresh, empty database with
                 // the current schema.
                 Err(open_database::Error::Corrupt)
             }
+            16 => {
+                clear_database(tx)?;
```

Comment:
Why do we need to clear those three tables here? How about other ones?

Reply:
That's a great question.  I was trying to copy the behavior of the old database code that dropped all tables.  I think the reasoning was that we probably want to re-ingest the suggestions after the schema upgrade.   That said, I don't think that's needed for this migration so I removed from the migration code and added a note to the "For any changes to the schema..." section.  WDYT?

PS:  I thought that deleting from those 3 tables should cause all of the data to be dropped since all other tables have a foreign key to them.   However, I notice that the `yelp_*` tables actually have a foreign key to `record_id` with an `ON DELETE CASCADE` clause.  How should those work?  Should I add `ON DELETE CASCADE` to those tables in the migration?

Reply:
Can you please help me understand the migration strategy here? I am a little confused as there appears to be two things we're handling here: 1). Switching from `cache_path` to `data_path`; 2). Adding a new `dismissed_suggestions` to the suggest DB.

Re: the path switch, are we planning to move the entire database to a new db under `data_path`? If so, shouldn't we just nuke the old database and create a new one, add the new dismissal table, and re-ingest everything?

As for yelp tables, unlike other suggestions, they don't follow the regular relational convention (`suggestions/keywords/icons`). We can just add all the yelp four tables to the drop list.

Reply:
The main goal of the PR is `1`, but I wanted to do `2` to make sure we had a good story on how to do migrations in the future.

> If so, shouldn't we just nuke the old database and create a new one, add the new dismissal table, and re-ingest everything?

I think that would work fine, but I'm not sure it's needed.  Adding the dismissal table shouldn't require re-ingestion AFAICT.  That said, in practice I think it will be the same since when a user upgrades their major version they'll hit one of the previous upgrades that nuked the DB.

I'm hoping that the `clear_database` function will be useful for future migrations.  There we are going to want to nuke most of the database, but keep the `dismissed_suggestions` and maybe other tables intact.

> As for yelp tables, unlike other suggestions, they don't follow the regular relational convention (suggestions/keywords/icons). We can just add all the yelp four tables to the drop list.

That makes sense, I'll update the `clear_database` to do that.

Reply:
Thanks for the clarification, it makes total sense to me now!